extensions/azcli/src/az.ts

```diff
@@ -247,35 +247,35 @@ function parseVersion(raw: string): string {
 	// ...
 	const start = raw.search('azure-cli');
 	const end = raw.search('core');
-	raw = raw.slice(start, end).replace('azure-cli', '');
+	raw = raw.slice(start, end).replace('azure-cli', '').replace('*', '');
```

Comment:
Add comment here explaining why you're replacing the * (including an example of the output in the comment is a good thing to do)

I'd still suggest a regex here for parsing out the version - that'd be a lot simpler way of handling this.